### PR TITLE
use `Map` for `targetQueues` and remove guid stuff

### PR DESCRIPTION
**Pull request by bekzod**
@rwjblue is this something that can be considered ?

### CODE REVIEW

**Comment by rwjblue** [comment]
Ya, definitely

File: lib/backburner/queue.ts [hunk 1]
```diff
@@ -155,30 +138,23 @@ export default class Queue {
...
+    let localQueue = this.targetQueues.get(target);
 
-    let index = findItem(target, method, queue);
-    if (index > -1) {
-      queue[index + 2] = args;  // replace args
-      queue[index + 3] = stack; // replace stack
+    if (localQueue === undefined) {
+      this.targetQueues.set(target, [
+        method,
+        this._queue.push(target, method, args, stack) - 4
```
Comment (rwjblue) [inline lib/backburner/queue.ts]: Can you remove this to a separate local variable? I know both of the original implementations did it this way, but (to me at least) having a separate variable named `indexIntoQueue` or something would make this a bit easier to grok.

File: tests/cancel-test.ts [hunk 2]
```diff
@@ -146,180 +146,6 @@ QUnit.test('cancel during flush', function(assert) {
   assert.ok(!functionWasCalled, 'function was not called');
 });
 
-QUnit.test('with GUID_KEY and target', function(assert) {
```
Comment (rwjblue) [inline tests/cancel-test.ts]: I don’t think all of these tests should be removed, one set of them should be refactored to not provide `peekGuid` or `GUID_KEY` but still confirm the expectation that the method is only called once.

Comment (bekzod) [reply tests/cancel-test.ts]: returned 2 tests back others are just duplicates

**Comment by bekzod** [comment]
would be nice to merge https://github.com/BackburnerJS/backburner.js/pull/287 first and release 1.3.2 then merge this or cherry pick it :P then go full with `Map` implementation

**Pull request merged**